src/pds/ingress/pds-nucleus-product-completion-checker.py

```diff
@@ -37,70 +59,173 @@ def lambda_handler(event, context):
         logger.error(f"Error processing S3 event: {event}. Exception: {str(e)}")
         raise e
 
-
 # Identifies and processes completed products
 def process_completed_products():
-    logger.info(f'Checking completed products')
+
+    logger.debug("Checking completed products")
 
     sql = """
-            select distinct s3_url_of_product_label from product where s3_url_of_product_label
+            select distinct s3_url_of_product_label from product where processing_status = 'INCOMPLETE' and s3_url_of_product_label
             NOT IN (select distinct s3_url_of_product_label  from product_data_file_mapping
             where s3_url_of_data_file
             NOT IN (select s3_url_of_data_file from data_file));
             """
 
     response = rds_data.execute_statement(
-        resourceArn=db_clust_arn,
-        secretArn=db_secret_arn,
-        database='pds_nucleus',
-        sql=sql)
+        resourceArn = db_clust_arn,
+        secretArn = db_secret_arn,
+        database = 'pds_nucleus',
+        sql = sql)
 
-    logger.info("Number of completed product labels: " + str(len(response['records'])))
+    logger.debug("Number of completed product labels : " + str(len(response['records'])))
+
+    n = 10
+    count = 0;
+    list_of_product_labels_to_process = []
 
     for record in response['records']:
+
+        count = count + 1
+
         for data_dict in record:
-            for data_type, data_value in data_dict.items():
-                update_product_processing_status_in_database(data_value, 'COMPLETE')
-                notify_completed_product(data_value)
-                time.sleep(1 / 100)
+            for data_type, s3_url_of_product_label in data_dict.items():
+                update_product_processing_status_in_database(s3_url_of_product_label,'COMPLETE')
+                list_of_product_labels_to_process.append(s3_url_of_product_label)
+
+
+        if count == n:
+            submit_data_to_nucleus(list_of_product_labels_to_process)
+            count = 0
+            list_of_product_labels_to_process = []
 
+    submit_data_to_nucleus(list_of_product_labels_to_process)
+    count = 0
+    list_of_product_labels_to_process = []
 
 # Updates the product processing status of the given s3_url_of_product_label
 def update_product_processing_status_in_database(s3_url_of_product_label, processing_status):
+
     sql = f"""
         UPDATE product
         SET processing_status = '{processing_status}'
-        # SET last_updated_epoch_time = {round(time.time() * 1000)}
+        # SET last_updated_epoch_time = {round(time.time()*1000)}
```

Comment:
String substitution into SQL makes me nervous since the strings `processing_status` and `s3_url_of_product_label` aren't sanitized. Possible SQL injection vulnerability. Could you use the `parameters` argument to `execute_statement` to pass in these values instead?

Reply:
@nutjob4life, Thanks for catching it! I updated the code to use parameters.